backend/src/controller/map.rs

```diff
@@ -0,0 +1,121 @@
+//! `Map` endpoints.
+
+use actix_web::web::Query;
+use actix_web::{
+    get, post,
+    web::{Data, Json, Path},
+    HttpResponse, Result,
+};
+
+use crate::model::dto::{
+    MapSearchParameters, MapVersionSearchParameters, NewMapVersionDto, PageParameters,
+};
+use crate::{db::connection::Pool, model::dto::NewMapDto, service};
+
+/// Endpoint for fetching or searching all [`Map`](crate::model::entity::Map).
+/// Search parameters are taken from the URLs query string (e.g. .../api/maps?is_inactive=false&per_page=5).
+/// If no page parameters are provided, the first page is returned.
+///
+/// # Errors
+/// * If the connection to the database could not be established.
+#[utoipa::path(
+    context_path = "/api/users/{user_id}/maps",
+    params(
+        MapSearchParameters,
+        PageParameters
+    ),
+    responses(
+        (status = 200, description = "Fetch or search all maps", body = PageMapDto)
+    )
+)]
+#[get("")]
+pub async fn find(
+    search_query: Query<MapSearchParameters>,
+    page_query: Query<PageParameters>,
+    pool: Data<Pool>,
+) -> Result<HttpResponse> {
+    let response =
+        service::map::find(search_query.into_inner(), page_query.into_inner(), &pool).await?;
+    Ok(HttpResponse::Ok().json(response))
+}
+
+/// Endpoint for fetching a [`Map`](crate::model::entity::Map).
+///
+/// # Errors
+/// * If the connection to the database could not be established.
+#[utoipa::path(
+    context_path = "/api/users/{user_id}/maps/{map_id}",
```

Comment:
Once auth works we might have to revise that (to either take the user_id directly from the token or at least validate it).
There is nothing to change here just something to remember.

Reply:
Thanks for telling us, can you please fix this in #330 (or create a follow-up issue).